pkg/server/apiutil/indexnames.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package apiutil
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+)
+
+// IndexNames and IndexNamesList are utilities for representing the indexes
+// found in descriptors in human readable formats.
+type IndexNames struct {
+	Database string
+	Table    string
+	Index    string
+	Span     roachpb.Span
+}
+
+func (idx IndexNames) String() string {
+	return fmt.Sprintf("%s.%s.%s", idx.Database, idx.Table, idx.Index)
+}
+
+type IndexNamesList []IndexNames
+
+// ToOutput is a simple function which returns a set of de-duplicated,
+// fully referenced database, table, and index names depending on the,
+// number of databases and tables which appear.
+func (idxl IndexNamesList) ToOutput() ([]string, []string, []string) {
+	seenDatabases := map[string]bool{}
+	databases := []string{}
+	for _, idx := range idxl {
+		if !seenDatabases[idx.Database] {
+			seenDatabases[idx.Database] = true
+			databases = append(databases, idx.Database)
+		}
+	}
+
+	multipleDatabases := len(databases) > 1
+	seenTables := map[string]bool{}
+	tables := []string{}
+	for _, idx := range idxl {
+		table := idx.Table
+		if multipleDatabases {
+			table = idx.Database + "." + table
```

Comment:
Are these strings quoted? Will we have any issues with db names and tables with `.` in them?
E.g. database `my.db` has table `table` and databas `my` has table `db.table`. Is it important that this output differentiates those?

Reply:
Have we run into this issue before? It can't hurt to quote them I suppose, though if it's not necessary I'd prefer against it.